pyocd/target/builtin/target_LPC55S36.py

```diff
@@ -0,0 +1,119 @@
+    0xbf987f00, 0x7500f44f, 0x0007eb09, 0x46224631, 0xf000462b, 0x2800f847, 0x2001bf18, 0xbdf0b001,
+    0x460cb5b0, 0xf0204605, 0x46114070, 0xf0004622, 0x2800f8b8, 0x4425bf08, 0xbdb04628, 0x460ab580,
+    0x4170f020, 0x000cf240, 0x0000f2c0, 0xf0004448, 0x2800f83f, 0x2001bf18, 0x0000bd80, 0x0108f240,
+    0x0100f2c0, 0xf8092201, 0xf64f2001, 0xf2c101dc, 0x68093102, 0xbf004708, 0x0c08f240, 0x0c00f2c0,
+    0xc00cf819, 0x0f00f1bc, 0xf64abf07, 0xf2c13c4f, 0xf64f3c00, 0xf2c10ce0, 0xbf183c02, 0xc000f8dc,
+    0xbf004760, 0x0c08f240, 0x0c00f2c0, 0xc00cf819, 0x0f00f1bc, 0xf248bf07, 0xf2c17c9b, 0xf64f3c02,
+    0xf2c10ce4, 0xbf183c02, 0xc000f8dc, 0xbf004760, 0x0308f240, 0x0300f2c0, 0x3003f819, 0xbf072b00,
+    0x3381f64a, 0x3300f2c1, 0x03e8f64f, 0x3302f2c1, 0x681bbf18, 0xbf004718, 0x0c08f240, 0x0c00f2c0,
+    0xc00cf819, 0x0f00f1bc, 0xf64abf07, 0xf2c14ca5, 0xf64f3c00, 0xf2c10cec, 0xbf183c02, 0xc000f8dc,
+    0xbf004760, 0x03f0f64f, 0x3302f2c1, 0x4718681b, 0x01f4f64f, 0x3102f2c1, 0x47086809, 0x01f8f64f,
+    0x3102f2c1, 0x47086809, 0x03fcf64f, 0x3302f2c1, 0x4718681b, 0x1c04f64f, 0x3c02f2c1, 0xc000f8dc,
+    0xbf004760, 0x1208f64f, 0x3202f2c1, 0x47106812, 0x120cf64f, 0x3202f2c1, 0x47106812, 0x1310f64f,
+    0x3302f2c1, 0x4718681b, 0x1200f64f, 0x3202f2c1, 0x47106812, 0x1c18f64f, 0x3c02f2c1, 0xc000f8dc,
+    0xea404760, 0xb5100301, 0xd10f079b, 0xd30d2a04, 0xc908c810, 0x429c1f12, 0xba20d0f8, 0x4288ba19,
+    0x2001d901, 0xf04fbd10, 0xbd1030ff, 0x07d3b11a, 0x1c52d003, 0x2000e007, 0xf810bd10, 0xf8113b01,
+    0x1b1b4b01, 0xf810d107, 0xf8113b01, 0x1b1b4b01, 0x1e92d101, 0x4618d1f1, 0x0000bd10, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000, 0x00000000,
+    0x00000000
+    ],
+
+    # Relative function addresses
+    'pc_init': 0x20000005,
+    'pc_unInit': 0x20000061,
+    'pc_program_page': 0x200000b5,
+    'pc_erase_sector': 0x2000008d,
+    'pc_eraseAll': 0x20000065,
+
+    'static_base' : 0x20000000 + 0x00000004 + 0x000002dc,
+    'begin_stack' : 0x20000500,
+    'begin_data' : 0x20000000 + 0x1000,
+    'page_size' : 0x200,
+    'analyzer_supported' : False,
+    'analyzer_address' : 0x00000000,
+    'page_buffers' : [0x20001000, 0x20001200],   # Enable double buffering
+    'min_program_length' : 0x200,
+
+    # Relative region addresses and sizes
+    'ro_start': 0x0,
+    'ro_size': 0x2dc,
+    'rw_start': 0x2dc,
+    'rw_size': 0x4,
+    'zi_start': 0x2e0,
+    'zi_size': 0x44,
+
+    # Flash information
+    'flash_start': 0x0,
+    'flash_size': 0x40000,
+    'sector_sizes': (
+        (0x0, 0x8000),
+    )
+}
+
+
+class LPC55S36(LPC5500Family):
+
+    MEMORY_MAP = MemoryMap(
+        FlashRegion(name='nsflash',     start=0x00000000, length=0x040000, access='rx',
+            page_size=0x200,
+            sector_size=0x8000,
```

Comment:
Fyi,  the sector size is actually 0x200 in the hardware, and the value in the DFP flash algo is incorrect. I verified this with the RM and testing. However, because the flash algo always erases 0x8000 we have to leave the sector size at that value.

And in fact, the flash algo, or at least the LPC55S69 version, has a bug that it will erase 0x8000 bytes started at whatever address you give it, even if that address is not aligned to 0x8000.

(I reported this to the SDK team via Jason Martin a while ago.)

Reply:
I will check with our team to fix it. Thanks flit!

Reply:
Thanks!